lib/segment/tests/integration/payload_index_test.rs

```diff
@@ -477,7 +477,7 @@ fn build_test_segments_nested_payload(path_struct: &Path, path_plain: &Path) ->
 
     for (_field, indexes) in struct_segment.payload_index.borrow().field_indexes.iter() {
         for index in indexes {
-            assert!(index.count_indexed_points() < num_points as usize);
+            assert!(index.count_indexed_points() <= num_points as usize);
```

Comment:
this is modified because now the null index is built for `on_disk=false` as well, and null index reports the max id that has had a value, even if it was deleted (`has_values_slice.len()`)